client/dashboard/section.js

```diff
@@ -3,16 +3,31 @@
  * External dependencies
  */
 import { Component } from '@wordpress/element';
+import { xor } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import DashboardCharts from './dashboard-charts';
+import Leaderboards from './leaderboards';
+import StorePerformance from './store-performance';
 
 export default class Section extends Component {
 	constructor( props ) {
 		super( props );
 		const { title } = props;
 
+		this.components = {
+			'store-performance': StorePerformance,
+			charts: DashboardCharts,
+			leaderboards: Leaderboards,
+		};
```

Comment:
This is going to pose a problem for extensions adding Sections.

Could you add the component somehow in this part of `applyFilters` in customizable.js?

```js
sections: applyFilters(
    'woocommerce_dashboard_sections',
    props.userPrefSections || defaultSections
),
```
A slightly less clean approach could be a `registerDashboardComponent` filter. What do you think?

Reply:
Good point. I refactored it in 1caa9d19f37cd96422be8d937c8ec9f8970f6763. Let me know how it looks.